cloudify_agent/api/plugins/installer.py

```diff
@@ -221,3 +262,89 @@ def extract_package_name(package_dir):
         cwd=package_dir
     ).std_out
     return plugin_name
+
+
+def get_plugin_id(plugin):
+    package_name = plugin.get('package_name')
```

Comment:
Is there currently a way for us to simply provide a url/blueprint-relative-path to a wagon archive for installation?
It would seem good to me to allow users to override the querying mechanism and just allow them to state: "this is the wagon I want for this plugin and I don't give a sh*t about anything else" :)